src/lib/datepicker/calendar.spec.ts

```diff
@@ -168,6 +168,31 @@ describe('MatCalendar', () => {
           expect(activeCell.focus).not.toHaveBeenCalled();
         });
 
+        it('should update today\'s date after update of the active date', () => {
+
+          // TODO: How can we actually check that today's date was updated correctly?
+          // TODO: The test only changes the active date, but not today's date.
+          // today's date is set by the data adapter's today() method
+
+          const newActiveDate = new Date();
+
+          // today's date is also the active date
+          let todaysDay = newActiveDate.getDate();
+
+          calendarInstance.activeDate = newActiveDate;
+          fixture.detectChanges();
+
+          calendarInstance._dateSelected(newActiveDate);
```

Comment:
I don't think so, this is only intended to be called when the user clicks on a date to select it.

Reply:
see <https://stackblitz.com/edit/angular-ncyrjg> `app.component.ts`

Reply:
Hmm I could see it making more sense to add the ability to set the selected date on the `mat-datepicker`, since that's how most developers would probably interact with it. Would that work for your case? You could probably inject the datepicker similar to how you injected the calendar (probably no `@Host` though, since datepicker is a few levels up)

Reply:
Thnaks for the hint, I will try that!

Reply:
Simply injecting it does not work:

> ERROR
Error: StaticInjectorError(AppModule)[HeaderComponent -> MatDatepicker]: 
StaticInjectorError(Platform: core)[HeaderComponent -> MatDatepicker]: 
NullInjectorError: No provider for MatDatepicker!

see <https://stackblitz.com/edit/angular-tx8iqu>

My first guess was that something was missing from `src/lib/datepicker/datepicker-module.ts`, but it seems ok to me:

<https://github.com/dhlab-basel/material2/blob/48dda505f78ba82be385b025c6c5eb5ff51e8a84/src/lib/datepicker/datepicker-module.ts#L31-L74>

However, this approach would work for the main component `AppComponent`:

```
@ViewChild('picker') private _datepicker: MatDatepicker<JDNConvertibleCalendar>;
```

But I need to access the `Datepicker` from `HeaderComponent`.

Reply:
Ah I see, it's because the datepicker is in opened in a popup which is outside of the actual datepicker element. What you can do is this:

```ts
constructor(..., private _datepickerContent: MatDatepickerCotnent<any>) {...}
```

and then `this._datepickerContent.datepicker` to get a hold of the actual datepicker. A bit round about, but should work.

Sorry for the delayed response, I was OOO for a while.

Reply:
No worries!

Yes, this works. You can see it here: <https://stackblitz.com/edit/angular-tx8iqu>

> Hmm I could see it making more sense to add the ability to set the selected date on the mat-datepicker, since that's how most developers would probably interact with it. 

Shall I add a public method to the datepicker that internally calls the private `_select` and then calls `updateTodaysDate` on `_calendar`?

Is this still needed? `this._calendar.activeDate = convertedDate`

Reply:
I think setting the active date on the calendar is still necessary. This could all be handled in one place when you call the new method on the datepicker.

Reply:
I don't know about automatically calling `updateTodaysDate`, I would leave that up to the user, but adding a public setter for the selected date should be fine.

Setting the `activeDate` is probably still necessary in your case. You'll want to update both the selected and the active after changing the calendar system.

Reply:
I think the combined method sounds kind of specific to your use case, not something that other developers would wind up needing that much, so I'd rather keep it separate in material and just have you call the various bits you need from your code

Reply:
Probably I did not understand the difference between the selected and the active date:

- selected date: date that user selects by clicking and is shown in the input field as text
- active date: date or date range that is currently shown to the user
- today's date: marks today's date with a circle

So if I get it correctly, the new public setter would just change the selected date, and the user would still have to care about showing it to the user by updating the active date and today's date.

Reply:
Ok, that does not seem hard to do :-)

How would we write a test for `updateTodaysDate`? Since we do not do calendar conversions here, today's date stays the same.